pkg/trace/stats/concentrator.go

```diff
@@ -46,8 +47,25 @@ type Concentrator struct {
 	agentEnv               string
 	agentHostname          string
 	agentVersion           string
-	peerSvcAggregation     bool // flag to enable peer.service aggregation
-	computeStatsBySpanKind bool // flag to enable computation of stats through checking the span.kind field
+	peerSvcAggregation     bool     // flag to enable peer.service aggregation
+	computeStatsBySpanKind bool     // flag to enable computation of stats through checking the span.kind field
+	peerTagKeys            []string // keys for supplementary tags that describe peer.service entities
+}
+
+func prepareTagKeys(tags ...string) []string {
+	if len(tags) == 0 {
+		return nil
+	}
+	ptm := make(map[string]struct{})
+	for _, t := range tags {
+		ptm[t] = struct{}{}
+	}
+	deduped := make([]string, 0, len(ptm))
+	for t := range ptm {
+		deduped = append(deduped, t)
+	}
+	sort.Strings(deduped)
```

Comment:
If you're sorting them here, then do you need to sort it in the other function earlier in this PR?

Reply:
I recommended sorting in the other function to defend against future programmers using the hash function incorrectly (without sorting the tags first). It would be a slight optimization to remove it, but we've mitigated the additional sort with first checking if the list is already sorted.

Reply:
+1 sounds good. Either way, let's not do it twice 😄

Reply:
Agreed - we should ultimately be sorting once. In the hash func we'll check to see we're sorted, which will be an extra O(n) cost, but we at least won't sort twice.